core/src/services/s3/backend.rs

```diff
@@ -667,10 +667,19 @@ impl S3Builder {
         self
     }
 
+    /// Check if `bucket` is valid
+    /// `bucket` must be not empty and if `enable_virtual_host_style` is true
+    /// it couldn't contain dot(.) character
+    fn is_valid_bucket(&self) -> bool {
+        !self.bucket.is_empty() &&
```

Comment:
I'm not a clever boy :rofl: and have difficulty understanding conditional expressions that contain more than one condition.

How about write them into:

```rust
// comment here, why
if self.bucket.is_empty() {
  return false
}
// comment here, why
if self.enable_virtual_host_style && self.bucket.contains(".") {
    return false
}

true
```

Reply:
> I'm not a clever boy

🤣 LMAO.